src/CommitmentsV2/SFA.DAS.CommitmentsV2/Services/AlertSummaryService.cs

```diff
@@ -0,0 +1,116 @@
+﻿using Microsoft.Extensions.Logging;
+using NServiceBus;
+using Polly;
+using Polly.Retry;
+using SFA.DAS.CommitmentsV2.Domain.Interfaces;
+using SFA.DAS.CommitmentsV2.Messages.Commands;
+using SFA.DAS.CommitmentsV2.Models;
+using SFA.DAS.CommitmentsV2.Models.ApprovalsOuterApi;
+using SFA.DAS.CommitmentsV2.Models.ApprovalsOuterApi.Types;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.CommitmentsV2.Services
+{
+    public class AlertSummaryService : IAlertSummaryService
+    {
+        private readonly IApprenticeshipDomainService _apprenticeshipDomainService;
+        private readonly IMessageSession _messageSession;
+        private readonly IApprovalsOuterApiClient _approvalsOuterApiClient;
+        private readonly ILogger<AlertSummaryService> _logger;
+        private readonly AsyncRetryPolicy _asyncRetryPolicy;
+
+        public AlertSummaryService(IApprenticeshipDomainService apprenticeshipDomainService, IMessageSession messageSession, IApprovalsOuterApiClient approvalsOuterApiClient, 
+            ILogger<AlertSummaryService> logger)
+        {
+            _apprenticeshipDomainService = apprenticeshipDomainService;
+            _messageSession = messageSession;
+            _approvalsOuterApiClient = approvalsOuterApiClient;
+            _logger = logger;
+            _asyncRetryPolicy = GetRetryPolicy();
+        }
+
+        public async Task SendEmployerAlertSummaryNotifications()
+        { 
+            var employerAlertSummaryNotifications = await _apprenticeshipDomainService.GetEmployerAlertSummaryNotifications();
+
+            var accountsTasks = employerAlertSummaryNotifications
+                .Select(x => _asyncRetryPolicy.ExecuteAndCaptureAsync(() => _approvalsOuterApiClient.Get<AccountResponse>(new GetAccountRequest(x.EmployerHashedAccountId))))
+                .ToList();
+
+            await Task.WhenAll(accountsTasks);
+
+            var accounts = accountsTasks
+                .Select(p => p.Result)
+                .Where(p => p.Outcome == OutcomeType.Successful)
+                .Select(p => p.Result)
+                .Where(p => p != null)
```

Comment:
Should this where come first?

Reply:
The first p is the task which is always either completed or not completed the second p is the result of the task which may be null.